integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/datamap/CarbonCreateDataMapCommand.scala

```diff
@@ -181,7 +181,18 @@ case class CarbonCreateDataMapCommand(
 
   override def undoMetadata(sparkSession: SparkSession, exception: Exception): Seq[Row] = {
     if (dataMapProvider != null) {
-      dataMapProvider.cleanMeta()
+      if (dataMapSchema.getProviderName
```

Comment:
I think datamap drop is required for all types of datamap , not just preaggregate

Reply:
We need this check because we want the implementation of cleanMeta() method in PreAggregregateDataMapProvider. For others, It will take from IndexDataMapProvider